src/libaktualizr/uptane/ipuptanesecondary.cc

```diff
@@ -54,21 +54,32 @@ bool IpUptaneSecondary::putMetadata(const RawMetaPack& meta_pack) {
   return r->result == AKInstallationResult_success;
 }
 
-bool IpUptaneSecondary::sendFirmware(const std::string& data) {
-  LOG_INFO << "Sending firmware` the secondary";
+bool IpUptaneSecondary::sendFirmwareAsync(const std::shared_ptr<std::string>& data) {
+  if (!install_future.valid() || install_future.wait_for(std::chrono::seconds(0)) == std::future_status::ready) {
+    install_future = std::async(std::launch::async, &IpUptaneSecondary::sendFirmware, this, std::ref(data));
```

Comment:
Sorry, I'm late but should it really send `std::ref` here? Looks like it will just pass around a reference to the shared_ptr, without incrementing its counter. So it could go out of scope inside `sendFirmware`

Reply:
@lbonn You were right. It actually crashed for me while testing. I'm fixing it in my PR.